### PR TITLE
Kanvi/bazel build with clang libc++

### PR DESCRIPTION
**Pull request by kanvi-nervana**
In order to build the `hello_tf` example with `clang` and `libc++`, this PR introduces a dockerfile which creates an env for the same. The `hello_tf` example cpp app is then built using the following command : `BAZEL_LINKOPTS=-lc++ BAZEL_CXXOPTS=-stdlib=libc++ bazel build :hello_tf` 
The `README` for the dockerfile gives the detail steps to be followed.

### CODE REVIEW

File: bazel/docker/Dockerfile.clang-8.ubuntu18.04 [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+
+#install latest cmake
+ADD https://cmake.org/files/v3.7/cmake-3.7.2-Linux-x86_64.sh /cmake-3.7.2-Linux-x86_64.sh
+RUN mkdir /opt/cmake
+RUN sh /cmake-3.7.2-Linux-x86_64.sh --prefix=/opt/cmake --skip-license
+RUN ln -s /opt/cmake/bin/cmake /usr/local/bin/cmake
+RUN cmake --version
+
+#Install Bazel 
+ADD https://github.com/bazelbuild/bazel/releases/download/0.24.1/bazel-0.24.1-installer-linux-x86_64.sh ./bazel-0.24.1-installer-linux-x86_64.sh
```
Comment (sayantan-nervana) [inline bazel/docker/Dockerfile.clang-8.ubuntu18.04]: Please update this wiki: https://github.com/tensorflow/ngraph-bridge/wiki/Update-TF-version-in-ngraph-bridge

when TF version is upgraded, bazel version will upgrade, and we need to upgrade this (and maybe other) docker file from 0.24.1 to next bazel version

File: bazel/docker/README.md [hunk 2]
```diff
@@ -0,0 +1,11 @@
+# How to use this Dockerfile (Dockerfile.clang-8.ubuntu18.04)
+1. Create a dir where you want to build eg: docker_demo/build
+2. Download ngraph-bridge repo : git clone https://github.com/tensorflow/ngraph-bridge.git
+3. docker build -t <ngtf_build> -f Dockerfile.clang-8.ubuntu18.04 .
+4. docker run -it -v ${PWD}:/<workspace> -w /<workspace> <ngtf_build>
+5. Create a virtual env : virtualenv -p python3 <venv3>
+6. Activate the virtual env : source venv3/bin/activate
+7. pip install future
```
Comment (sayantan-nervana) [inline bazel/docker/README.md]: Just making sure. There are 2 python packages, `future` and `futures`. We want `future` here, right?

Comment (kanvi-nervana) [reply bazel/docker/README.md]: Yes

File: bazel/WORKSPACE [hunk 3]
```diff
@@ -55,11 +55,11 @@ tf_workspace(path_prefix = "", tf_repo_name = "org_tensorflow")
 http_archive(
     name = "ngraph",
     build_file = "//:bazel/ngraph.BUILD",
-    sha256 = "b33f0be032ff59044987ff796b636c1b222579705ac29632bba59b6031b0d58c",
-    strip_prefix = "ngraph-0.25.1-rc.3",
+    sha256 = "2437295ae696cee472231c38149cfa7cc46b6266c101ef76693e4519cb0604d7",
+    strip_prefix = "ngraph-0.25.1-rc.5",
```
Comment (avijit-nervana) [inline bazel/WORKSPACE]: nGraph needs to be in the latest version i.e., rc7

File: bazel/docker/Dockerfile.clang-8.ubuntu18.04 [hunk 4]
```diff
@@ -0,0 +1,58 @@
...
+#  Unless required by applicable law or agreed to in writing, software
+#  distributed under the License is distributed on an "AS IS" BASIS,
+#  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#  See the License for the specific language governing permissions and
+#  limitations under the License.
+# ==============================================================================
+# Docker file to setup build envrionment for nGraph-TensorFlow
+
+FROM ubuntu:18.04
+ENV http_proxy 'http://proxy-chain.intel.com:911'
```
Comment (avijit-nervana) [inline bazel/docker/Dockerfile.clang-8.ubuntu18.04]: Remove these two lines. Update the instructions with information from here: https://github.com/tensorflow/ngraph-bridge/pull/212#issue-308729569

File: bazel/ngraph.BUILD [hunk 5]
```diff
@@ -84,7 +84,7 @@ cc_library(
         "-fstack-protector-all",
         '-D SHARED_LIB_PREFIX=\\"lib\\"',
         '-D SHARED_LIB_SUFFIX=\\".so\\"',
-        '-D NGRAPH_VERSION=\\"v0.25.1-rc.3\\"',
+        '-D NGRAPH_VERSION=\\"v0.25.1-rc.5\\"',
```
Comment (avijit-nervana) [inline bazel/ngraph.BUILD]: Change the version

**Pull request merged**